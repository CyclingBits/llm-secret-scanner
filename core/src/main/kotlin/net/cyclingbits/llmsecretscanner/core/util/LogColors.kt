package net.cyclingbits.llmsecretscanner.core.util

object LogColors {
    // ANSI Color codes
    const val RESET = "\u001B[0m"
    const val RED = "\u001B[31m"
    const val GREEN = "\u001B[32m"
    const val YELLOW = "\u001B[33m"
    const val BLUE = "\u001B[34m"
    const val PURPLE = "\u001B[35m"
    const val CYAN = "\u001B[36m"
    const val WHITE = "\u001B[37m"
    const val BRIGHT_GREEN = "\u001B[92m"
    const val BRIGHT_YELLOW = "\u001B[93m"
    const val BRIGHT_BLUE = "\u001B[94m"
    const val BRIGHT_CYAN = "\u001B[96m"
    
    // Bold colors
    const val BOLD_RED = "\u001B[1;31m"
    const val BOLD_GREEN = "\u001B[1;32m"
    const val BOLD_YELLOW = "\u001B[1;33m"
    const val BOLD_BLUE = "\u001B[1;34m"
    const val BOLD_CYAN = "\u001B[1;36m"
    
    // Icons/Emojis
    const val SCANNER_ICON = "üîç"
    const val SUCCESS_ICON = "‚úÖ"
    const val WARNING_ICON = "‚ö†Ô∏è"
    const val ERROR_ICON = "‚ùå"
    const val INFO_ICON = "‚ÑπÔ∏è"
    const val DOCKER_ICON = "üê≥"
    const val FILE_ICON = "üìÑ"
    const val SECURITY_ICON = "üîí"
    const val ISSUE_ICON = "üö®"
    
    // Helper functions
    fun red(text: String) = "$RED$text$RESET"
    fun green(text: String) = "$GREEN$text$RESET"
    fun yellow(text: String) = "$YELLOW$text$RESET"
    fun blue(text: String) = "$BLUE$text$RESET"
    fun cyan(text: String) = "$CYAN$text$RESET"
    fun boldRed(text: String) = "$BOLD_RED$text$RESET"
    fun boldGreen(text: String) = "$BOLD_GREEN$text$RESET"
    fun boldYellow(text: String) = "$BOLD_YELLOW$text$RESET"
    fun boldBlue(text: String) = "$BOLD_BLUE$text$RESET"
    fun boldCyan(text: String) = "$BOLD_CYAN$text$RESET"
}