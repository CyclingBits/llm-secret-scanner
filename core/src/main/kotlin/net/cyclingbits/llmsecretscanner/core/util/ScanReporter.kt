package net.cyclingbits.llmsecretscanner.core.util

import net.cyclingbits.llmsecretscanner.core.config.ScannerConfiguration
import net.cyclingbits.llmsecretscanner.core.config.ScannerDefaults
import net.cyclingbits.llmsecretscanner.core.model.Issue
import org.slf4j.LoggerFactory
import java.io.File

object ScanReporter {

    private val logger = LoggerFactory.getLogger(ScanReporter::class.java)

    private fun findBaseDirectory(file: File, sourceDirectories: List<File>): File {
        return sourceDirectories.find { dir ->
            file.absolutePath.startsWith(dir.absolutePath)
        } ?: sourceDirectories.firstOrNull() ?: File(".")
    }

    fun reportScanStart(config: ScannerConfiguration) {
        logger.info("")
        logger.info("{} {}", LogColors.SCANNER_ICON, LogColors.boldCyan("Starting LLM Secret Scanner with configuration:"))
        if (config.sourceDirectories.size == 1) {
            logger.info("       {} Source directory: {}", LogColors.FILE_ICON, LogColors.blue(config.sourceDirectories[0].absolutePath))
        } else {
            logger.info("       {} Source directories ({}):", LogColors.FILE_ICON, LogColors.yellow(config.sourceDirectories.size.toString()))
            config.sourceDirectories.forEach { dir ->
                logger.info("         - {}", LogColors.blue(dir.absolutePath))
            }
        }
        logger.info("       {} Model: {}", LogColors.DOCKER_ICON, LogColors.green(config.modelName))
        logger.info("       ‚è±Ô∏è Chunk analysis timeout: {}s", LogColors.yellow(config.chunkAnalysisTimeout.toString()))
        logger.info("       ‚úÖ Include patterns: {}", LogColors.cyan(config.includes))
        logger.info("       ‚ùå Exclude patterns: {}", LogColors.yellow(config.excludes))
    }

    fun reportFilesFound(fileCount: Int) {
        logger.info("{} Found {} files matching patterns", LogColors.FILE_ICON, LogColors.boldYellow(fileCount.toString()))
    }

    fun reportNoFilesFound() {
        logger.warn("No files found matching the specified patterns")
    }

    fun reportScanComplete(totalIssues: Int, filesAnalyzed: Int, totalFiles: Int, totalTimeMs: Long) {
        logger.info("")
        val totalTimeSeconds = String.format("%.1f", totalTimeMs / 1000.0)
        logger.info(
            "{} {} Analyzed {} of {} files, found {} total issues in {}s",
            LogColors.SUCCESS_ICON,
            LogColors.boldGreen("Scan completed successfully."),
            LogColors.boldCyan(filesAnalyzed.toString()),
            LogColors.boldYellow(totalFiles.toString()),
            LogColors.boldYellow(totalIssues.toString()),
            LogColors.yellow(totalTimeSeconds)
        )
    }

    fun reportError(message: String) {
        logger.error("{} {}", LogColors.ERROR_ICON, LogColors.boldRed(message))
    }

    fun reportError(message: String, exception: Throwable) {
        logger.error("{} {}", LogColors.ERROR_ICON, LogColors.boldRed(message), exception)
    }

    fun reportWarning(message: String) {
        logger.warn("       ‚ö†Ô∏è {}", LogColors.yellow(message))
    }

    fun reportChunk(chunkCount: Int, maxLinesPerChunk: Int, overlapLines: Int) {
        logger.info("       üìÑ Chunking file into {} chunks ({} lines per chunk, {} overlap)", 
            LogColors.boldYellow(chunkCount.toString()),
            LogColors.cyan(maxLinesPerChunk.toString()),
            LogColors.cyan(overlapLines.toString())
        )
    }

    fun reportChunkAnalysis(chunkIndex: Int, totalChunks: Int, startLine: Int, endLine: Int) {
        logger.info("       üîç Analyzing chunk [{}/{}] (lines {}-{})", 
            LogColors.cyan(chunkIndex.toString()),
            LogColors.cyan(totalChunks.toString()),
            LogColors.blue(startLine.toString()),
            LogColors.blue(endLine.toString())
        )
    }

    fun reportContainerStart() {
        logger.info("")
        logger.info("{} Starting Docker container...", LogColors.DOCKER_ICON)
    }

    fun reportContainerStarted() {
        logger.info("{} Docker container started successfully", LogColors.SUCCESS_ICON)
    }

    fun reportAnalysisStartForDirectory(directory: File, fileCount: Int) {
        logger.info("")
        logger.info("{} Starting analysis of {} files in {}",
            LogColors.SCANNER_ICON, 
            LogColors.boldYellow(fileCount.toString()),
            LogColors.blue(directory.absolutePath)
        )
    }

    fun reportFileAnalysisStart(file: File, fileIndex: Int, totalFiles: Int, sourceDirectories: List<File>) {
        val baseDir = findBaseDirectory(file, sourceDirectories)
        logger.info("")
        val relativePath = try {
            baseDir.toPath().relativize(file.toPath()).toString()
        } catch (e: IllegalArgumentException) {
            file.name
        }
        logger.info(
            "[{}/{}] Analyzing file {}...",
            LogColors.cyan(fileIndex.toString()),
            LogColors.cyan(totalFiles.toString()),
            LogColors.blue(relativePath)
        )
    }

    fun reportFileIssues(file: File, issues: List<Issue>, analysisTimeMs: Long, fileIndex: Int, totalFiles: Int, sourceDirectories: List<File>) {
        val baseDir = findBaseDirectory(file, sourceDirectories)
        val analysisTimeSeconds = String.format("%.1f", analysisTimeMs / 1000.0)
        logger.info(
            "       Found {} {} (analyzed in {}s)",
            LogColors.boldYellow(issues.size.toString()),
            LogColors.cyan("issues"),
            LogColors.yellow(analysisTimeSeconds)
        )

        issues.forEachIndexed { index, issue ->
            val issueColor = when (issue.issueType.lowercase()) {
                "password" -> LogColors.boldRed(issue.issueType)
                "api key", "token" -> LogColors.boldYellow(issue.issueType)
                "database credentials" -> LogColors.boldRed(issue.issueType)
                "private key" -> LogColors.boldRed(issue.issueType)
                "secret" -> LogColors.boldYellow(issue.issueType)
                else -> LogColors.cyan(issue.issueType)
            }

            logger.info(
                "       {} #{} | Line {} | {} | {}",
                LogColors.ISSUE_ICON,
                LogColors.yellow((index + 1).toString()),
                LogColors.blue(issue.lineNumber.toString()),
                issueColor,
                (issue.secretValue ?: "No secret value").let { if (it.length > ScannerDefaults.SECRET_DISPLAY_LENGTH) it.take(ScannerDefaults.SECRET_DISPLAY_LENGTH) + "..." else it }
            )
        }
    }

    fun reportCorrectDetection(issue: Issue) {
        logger.info("{} Correct at line {} - {}", 
            LogColors.SUCCESS_ICON, 
            LogColors.blue(issue.lineNumber.toString()), 
            issue.secretValue?.take(ScannerDefaults.SECRET_DISPLAY_LENGTH) ?: "No value"
        )
    }

    fun reportIncorrectDetection(issue: Issue) {
        logger.info("{} Incorrect at line {} - {}", 
            LogColors.ERROR_ICON, 
            LogColors.blue(issue.lineNumber.toString()), 
            issue.secretValue?.take(ScannerDefaults.SECRET_DISPLAY_LENGTH) ?: "No value"
        )
    }

    fun reportMissedSecret(issue: Issue) {
        logger.info("{} Missed at line {} - {}", 
            LogColors.ERROR_ICON, 
            LogColors.blue(issue.lineNumber.toString()), 
            issue.secretValue?.take(ScannerDefaults.SECRET_DISPLAY_LENGTH) ?: "No value"
        )
    }

    fun reportEvaluationStart(message: String) {
        logger.info("")
        logger.info("{} {}", LogColors.SCANNER_ICON, LogColors.boldCyan(message))
    }

    fun reportEvaluationComplete(message: String) {
        logger.info("{} {}", LogColors.SUCCESS_ICON, LogColors.boldGreen(message))
    }

    fun reportFileSaved(filePath: String) {
        logger.info("{} Results saved to {}", "üíæ", LogColors.blue(filePath))
    }

    fun reportDetectionResults(correctCount: Int, incorrectCount: Int, missedCount: Int) {
        logger.info("")
        logger.info("üìä Detection results: {} correct, {} incorrect, {} missed",
            LogColors.boldGreen(correctCount.toString()), 
            LogColors.boldRed(incorrectCount.toString()),
            LogColors.boldYellow(missedCount.toString())
        )
    }

    fun reportFalsePositives(falsePositiveCount: Int) {
        logger.info("‚ö†Ô∏è False positives: {} issues detected in files that should have none",
            LogColors.boldRed(falsePositiveCount.toString())
        )
    }
}