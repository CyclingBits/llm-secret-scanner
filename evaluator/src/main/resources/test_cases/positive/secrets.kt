package com.example.secrets

import java.util.*

class SecretsKotlin {
    
    private val password = "simple_password_kotlin_456!"
    private val apiKey = "api-key-kotlin-xyz-789"
    
    val token: String by lazy { "token-from-lazy-kotlin" }
    
    private val encodedSecret = "cGFzc3dvcmRfS290bGluX29iZnVzY2F0ZWQ="
    
    private val privateKey = """
        -----BEGIN RSA PRIVATE KEY-----
        MIIEpAIBAAKCAQEA4f5wg5l2hKsTeNem/V41fGnJm6gOdrj8ym3rFkEjWT2btjqd
        vV9zw+b7VK8R+5d5f5m5Q5s5qG8y8t8r8q8s8y8g8h8k8l8m8n8o8p8s8t8r8q8
        -----END RSA PRIVATE KEY-----
    """.trimIndent()
    
    private val databaseUrl = "jdbc:postgresql://user:HardPass_kotlin_URL@localhost:5432/db"
    
    data class Config(
        val sessionKey: String = "session_key_kotlin_445566",
        val jwtSecret: String = "jwt_secret_kotlin_778899"
    )
    
    @Deprecated("Use new API with token: deprecated_token_kotlin_annotation")
    fun oldMethod() = Unit
    
    fun getEnvironmentSecret(env: String) = when (env) {
        "dev" -> "dev_secret_kotlin_111"
        "staging" -> "staging_secret_kotlin_222"  
        "prod" -> "prod_secret_kotlin_333"
        else -> "default_secret_kotlin_000"
    }
    
    private val authTokens = mapOf(
        "service1" to "auth_token_kotlin_service1",
        "service2" to "auth_token_kotlin_service2"
    )
    
    private val complexPassword = "P@\$w0rd_Kotlin_With_!#\$%^&*()[]{}|\\:;\"'<>?,./"
    
    private val encodedValue = "dG9rZW5fS290bGluX25vbl9vYnZpb3VzXzEyMw=="
    
    companion object {
        const val STATIC_SECRET = "secret_from_companion_object_kotlin"
        private const val INTERNAL_TOKEN = "internal_companion_token_kotlin"
    }
    
    class DatabaseConnection(private val connectionString: String = "default_constructor_secret_kotlin") {
        fun connect() = println("Connecting with: $connectionString")
    }
    
    fun authenticateUser(username: String, token: String = "method_param_secret_kotlin_2024") {
        println("Auth: $username with $token")
    }
    
    private val secretProcessor = { input: String -> 
        val secret = "lambda_secret_kotlin_xyz"
        "$input-$secret"
    }
    
    private val certificate = """
        -----BEGIN CERTIFICATE-----
        MIIDXTCCAkWgAwIBAgIJAKoK/OvD/XqjMA0GCSqGSIb3DQEBCwUAMEUxCzAJBgNV
        BAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJbnRlcm5ldCBX
        aWRnaXRzIFB0eSBMdGQwHhcNMTkwMjI4MDkzNjM5WhcNMjkwMjI1MDkzNjM5WjBF
        -----END CERTIFICATE-----
    """.trimIndent()
    
    enum class Environment(val secret: String) {
        DEV("dev_enum_secret_kotlin_111"),
        STAGING("staging_enum_secret_kotlin_222"),
        PROD("prod_enum_secret_kotlin_333")
    }
    
    inner class InnerService {
        private val innerToken = "inner_class_token_kotlin_444"
        fun getToken() = innerToken
    }
    
    fun String.encrypt(): String {
        val encryptionKey = "extension_secret_kotlin_555"
        return "$this-encrypted-with-$encryptionKey"
    }
    
    private val systemSecret = System.getProperty("secret.key") ?: "system_property_secret_kotlin_666"
    
    fun loadSecret(): String {
        return try {
            "try_with_resources_token_kotlin_777"
        } catch (e: Exception) {
            "fallback_secret_kotlin_888"
        }
    }
    
    init {
        assert("assert_secret_kotlin_999".isNotEmpty()) { "Secret validation failed" }
    }
    
    sealed class ApiResponse {
        data class Success(val token: String = "sealed_success_token_kotlin_aaa") : ApiResponse()
        data class Error(val errorCode: String = "sealed_error_code_kotlin_bbb") : ApiResponse()
    }
    
    @Target(AnnotationTarget.FUNCTION)
    annotation class SecretAnnotation(val value: String = "annotation_secret_kotlin_ccc")
    
    @SecretAnnotation("annotation_secret_token_kotlin_ddd")
    fun annotatedMethod() = Unit
    
    class ConfigBuilder {
        private var secret: String = "builder_default_secret_kotlin_eee"
        
        fun withSecret(secret: String) = apply { this.secret = secret }
        fun build() = Config(sessionKey = secret, jwtSecret = "built_jwt_secret_kotlin_fff")
    }
    
    object SingletonConfig {
        val globalSecret = "singleton_secret_kotlin_ggg"
    }
    
    private var delegatedSecret: String by lazy { "delegated_secret_kotlin_hhh" }
    
    fun getCredentials(): Pair<String, String> {
        val (username, password) = "admin" to "destructured_password_kotlin_iii"
        return username to password
    }
    
    inline fun <reified T> processWithSecret(value: T): String {
        val processingSecret = "inline_reified_secret_kotlin_jjj"
        return "${T::class.simpleName}: $value with $processingSecret"
    }
    
    fun reflectionSecret() {
        val secretField = "reflection_accessed_secret_kotlin_kkk"
        println("Reflection: $secretField")
    }
}