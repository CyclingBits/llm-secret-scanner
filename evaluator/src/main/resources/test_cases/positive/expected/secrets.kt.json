[
  {
    "filePath": "secrets.kt",
    "issueNumber": 1,
    "lineNumber": 7,
    "secretValue": "simple_password_kotlin_456!"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 2,
    "lineNumber": 8,
    "secretValue": "api-key-kotlin-xyz-789"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 3,
    "lineNumber": 10,
    "secretValue": "token-from-lazy-kotlin"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 4,
    "lineNumber": 12,
    "secretValue": "cGFzc3dvcmRfS290bGluX29iZnVzY2F0ZWQ="
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 5,
    "lineNumber": 14,
    "secretValue": "-----BEGIN RSA PRIVATE KEY-----\n        MIIEpAIBAAKCAQEA4f5wg5l2hKsTeNem/V41fGnJm6gOdrj8ym3rFkEjWT2btjqd\n        vV9zw+b7VK8R+5d5f5m5Q5s5qG8y8t8r8q8s8y8g8h8k8l8m8n8o8p8s8t8r8q8\n        -----END RSA PRIVATE KEY-----"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 6,
    "lineNumber": 21,
    "secretValue": "jdbc:postgresql://user:HardPass_kotlin_URL@localhost:5432/db"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 7,
    "lineNumber": 24,
    "secretValue": "session_key_kotlin_445566"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 8,
    "lineNumber": 25,
    "secretValue": "jwt_secret_kotlin_778899"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 9,
    "lineNumber": 28,
    "secretValue": "deprecated_token_kotlin_annotation"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 10,
    "lineNumber": 32,
    "secretValue": "dev_secret_kotlin_111"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 11,
    "lineNumber": 33,
    "secretValue": "staging_secret_kotlin_222"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 12,
    "lineNumber": 34,
    "secretValue": "prod_secret_kotlin_333"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 13,
    "lineNumber": 35,
    "secretValue": "default_secret_kotlin_000"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 14,
    "lineNumber": 39,
    "secretValue": "auth_token_kotlin_service1"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 15,
    "lineNumber": 40,
    "secretValue": "auth_token_kotlin_service2"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 16,
    "lineNumber": 43,
    "secretValue": "P@$w0rd_Kotlin_With_!#$%^&*()[]{}|\\:;\"'<>?,./"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 17,
    "lineNumber": 45,
    "secretValue": "dG9rZW5fS290bGluX25vbl9vYnZpb3VzXzEyMw=="
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 18,
    "lineNumber": 48,
    "secretValue": "secret_from_companion_object_kotlin"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 19,
    "lineNumber": 49,
    "secretValue": "internal_companion_token_kotlin"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 20,
    "lineNumber": 52,
    "secretValue": "default_constructor_secret_kotlin"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 21,
    "lineNumber": 56,
    "secretValue": "method_param_secret_kotlin_2024"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 22,
    "lineNumber": 61,
    "secretValue": "lambda_secret_kotlin_xyz"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 23,
    "lineNumber": 65,
    "secretValue": "-----BEGIN CERTIFICATE-----\\n...\\n-----END CERTIFICATE-----"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 24,
    "lineNumber": 74,
    "secretValue": "dev_enum_secret_kotlin_111"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 25,
    "lineNumber": 75,
    "secretValue": "staging_enum_secret_kotlin_222"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 26,
    "lineNumber": 76,
    "secretValue": "prod_enum_secret_kotlin_333"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 27,
    "lineNumber": 80,
    "secretValue": "inner_class_token_kotlin_444"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 28,
    "lineNumber": 85,
    "secretValue": "extension_secret_kotlin_555"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 29,
    "lineNumber": 89,
    "secretValue": "system_property_secret_kotlin_666"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 30,
    "lineNumber": 93,
    "secretValue": "try_with_resources_token_kotlin_777"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 31,
    "lineNumber": 95,
    "secretValue": "fallback_secret_kotlin_888"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 32,
    "lineNumber": 100,
    "secretValue": "assert_secret_kotlin_999"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 33,
    "lineNumber": 104,
    "secretValue": "sealed_success_token_kotlin_aaa"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 34,
    "lineNumber": 105,
    "secretValue": "sealed_error_code_kotlin_bbb"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 35,
    "lineNumber": 109,
    "secretValue": "annotation_secret_kotlin_ccc"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 36,
    "lineNumber": 111,
    "secretValue": "annotation_secret_token_kotlin_ddd"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 37,
    "lineNumber": 115,
    "secretValue": "builder_default_secret_kotlin_eee"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 38,
    "lineNumber": 118,
    "secretValue": "built_jwt_secret_kotlin_fff"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 39,
    "lineNumber": 122,
    "secretValue": "singleton_secret_kotlin_ggg"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 40,
    "lineNumber": 125,
    "secretValue": "delegated_secret_kotlin_hhh"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 41,
    "lineNumber": 128,
    "secretValue": "destructured_password_kotlin_iii"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 42,
    "lineNumber": 133,
    "secretValue": "inline_reified_secret_kotlin_jjj"
  },
  {
    "filePath": "secrets.kt",
    "issueNumber": 43,
    "lineNumber": 138,
    "secretValue": "reflection_accessed_secret_kotlin_kkk"
  }
]
